### DISCUSSION TITLE
Why do I get an uncontrolled error for my text field from Material UI in my BasicData component?

### DISCUSSION THREAD

**Discussion by REDLIGHTFILMS** [react-hook-form/react-hook-form#5801] [Q&A] [answered]
I have the following problem. I always get an uncontrolled error, but I do not know why?

my code:

BasicData component: 

import React from "react";
import { Controller, useFormContext } from "react-hook-form";
import { TextField } from "@material-ui/core";

interface IProps {}

```
const BasicData: React.FC<IProps> = () => {
  const { control } = useFormContext();
  return (
    <div style={{ paddingTop: 20, paddingBottom: 40, paddingLeft: 5 }}>
      <Controller
        control={control}
        name="test.firstname"
        rules={{ required: true }}
        render={({ field }) => (
          <TextField
            label="First Name"
            variant="outlined"
            placeholder="Enter Your First Name"
            {...field}
          />
        )}
      />
    </div>
  );
};

export default BasicData;
```
my codesandbox: https://codesandbox.io/s/stoic-frost-lpnwe

**Comment by bluebill1049** [comment, marked as answer]
your code looks fine to me, but the codesandbox has a lot of components, can you do a isolated one?

**Reply by REDLIGHTFILMS** [reply]
I have isolated the other components. I commented out the stepper and wrote the BasicData component under it. But unfortunately the problem still occurs. 

https://codesandbox.io/s/stoic-frost-lpnwe

**Reply by REDLIGHTFILMS** [reply]
I solved my problem :) It was due to the defaultValue,
